### PR TITLE
Add PWM service

### PR DESCRIPTION
**Pull request by josiah-tesfu**
Implemented the PWM module for the RPC Javascript client. The implementation mirrors the PWM module in the Python RPC server. Included integration and unit tests.

This client module interfaces with the Python EdgePi RPC server through protobuf. It acts as a client for the Node-RED PWM module (not yet published).

Moved files inside the `serviceTypes` directory into a single file. This simplifies imports.

### CODE REVIEW

File: src/services/PwmService/PwmEnums.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import path from 'path'
+import * as protobuf from 'protobufjs'
+import type {PwmMap, PolarityMap} from './PwmTypes'
+
+const protoPckgPath = path.join(require.resolve('@edgepi-cloud/edgepi-rpc-protobuf'), '../edgepi_rpc_protos');
+const root = protobuf.loadSync(path.join(protoPckgPath, 'pwm.proto'))
+
+export const PWMPins = root.lookupEnum('PWMPins').values as unknown as PwmMap
+export const Polarity = root.lookupEnum('Polarity').values as unknown as PolarityMap
```
Comment (naomiven) [inline src/services/PwmService/PwmEnums.ts:9]: what does `as unknown as PwmMap` do?

**Review by naomiven — COMMENTED** [review]
looks good

File: .DS_Store
Comment (farzadpanahi) [inline .DS_Store:1]: why? please pay attention to what is being added. review PR files yourself!

File: src/.DS_Store
Comment (farzadpanahi) [inline src/.DS_Store:1]: why!

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@edgepi-cloud/edgepi-rpc",
-  "version": "1.0.12",
+  "version": "1.0.13",
```
Comment (farzadpanahi) [inline package.json:3]: let's change it to `1.1.0` since we are adding a new feature.

File: tests/.DS_Store
Comment (farzadpanahi) [inline tests/.DS_Store:1]: why!

File: tests/integration/PwmService.test.ts
Comment (farzadpanahi) [inline tests/integration/PwmService.test.ts:1]: add test for setting values and then verifying set values.

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: src/services/PwmService/PwmService.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    const response: serverResponse = await this.rpcChannel.callMethod(
+      serviceReq,
+      requestType,
+      responseType
+    );
+
+    if (response.error !== undefined) {
+      throw Error(response.error);
+    }
```
Comment (flavius-t) [inline src/services/PwmService/PwmService.ts:178]: Is it possible for response to be null after the call to `this.rpcChannel.callMethod`? If so, wouldn't checking `response.error` raise an error?

File: src/services/PwmService/PwmService.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  /**
+   * @async Sends an RPC request for a specified PWM SDK method
+   */
+  private async PWMRequest(
+    methodName: string,
+    pwmNum: PWM,
+    responseTypeName: string,
+  ): Promise<serverResponse> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.PWM"
```
Comment (flavius-t) [inline src/services/PwmService/PwmService.ts:135]: Is it possible to use constants instead for these string args?

**Review by flavius-t — CHANGES_REQUESTED** [review]

File: src/services/PwmService/PwmService.ts (hunk 4 above)
Comment (josiah-tesfu) [reply src/services/PwmService/PwmService.ts]: Yeah, I think that would work well. Updated to use constants.

File: src/services/PwmService/PwmService.ts (hunk 3 above)
Comment (josiah-tesfu) [reply src/services/PwmService/PwmService.ts]: You're right! I updated to check for that.

File: src/services/PwmService/PwmEnums.ts (hunk 1 above)
Comment (josiah-tesfu) [reply src/services/PwmService/PwmEnums.ts]: Allows easier type assertions for TypeScript. `as unknown` is an intermediary step that allows the type to be converted to the custom type `PwmMap` defined in PwmTypes.ts.

**Pull request closed without merging**